src/dsp_tools/models/connection.py

```diff
@@ -27,240 +29,260 @@ def check_for_api_error(response: requests.Response) -> None:
-                self._server + "/v2/authentication",
-                headers={"Authorization": "Bearer " + self._token},
+                self.server + "/v2/authentication",
+                headers={"Authorization": "Bearer " + self.token},
                 timeout=20,
             )
             check_for_api_error(response)
-            self._token = None
-
-    def __del__(self):
-        pass
-        # self.logout()
-
-    def post(self, path: str, jsondata: Optional[str] = None):
+            self.token = None
+
+    def write_log_file(
+        self,
+        method: str,
+        route: str,
+        headers: dict[str, str],
+        jsondata: Optional[str],
+        params: Optional[dict[str, Any]],
+        response: requests.Response,
+    ) -> None:
         """
-        Post Json data to a given server using a HTTP POST request
-        :param path: Path of RESTful route
-        :param jsondata: Valid JSON as string
-        :return: Response from server
+        If logging is enabled (self.log == True), write the request and response to a file.
+
+        Args:
+            method: HTTP method (POST, GET, PUT, DELETE)
+            route: route of DSP-API that was called
+            headers: headers of the HTTP request
+            jsondata: data sent to the server
+            response: response of the server
         """
+        if not self.log:
+            return
+        if response.status_code == 200:  # pylint: disable=duplicate-code
+            _return = response.json()
+        else:
+            _return = {"status": response.status_code, "message": response.text}
+        logobj = {
+            "DSP server": self.server,
+            "route": route,
+            "method": method,
+            "headers": headers,
+            "params": params,
+            "body": json.loads(jsondata) if jsondata else None,
+            "return-headers": dict(response.headers),
+            "return": _return,
+        }
+        filename = f"{datetime.now().strftime('%Y-%m-%d %H.%M.%S.%f')} {method} {route.replace('/', '_')}.json"
+        with open(self.log_directory / filename, "w", encoding="utf-8") as f:
+            json.dump(logobj, f, indent=4)
+
+    def post(
+        self,
+        path: str,
+        jsondata: Optional[str] = None,
+    ) -> dict[str, Any]:
+        """
+        Make a HTTP POST request to the server to which this connection has been established.
+
+        Args:
+            path: Path of RESTful route
+            jsondata: Valid JSON as string
 
+        Returns:
+            response from server
+        """
         # timeout must be None,
         # otherwise the client can get a timeout error while the API is still processing the request
         # in that case, the client's retry will fail, and the response of the original API call is lost
         timeout = None
-
-        if path[0] != "/":
+        if not path.startswith("/"):
             path = "/" + path
```

Comment:
I'm not sure under what condition this can happen. But could it be fixed on the caller side?

Reply:
That's a simple precaution measure, if one of the (many!) callers forgets the slash. Of course I could remove it, and fix all the callers. But isn't it a nice little feature that makes usage easier?